src/app/Http/Controllers/CrudController.php

```diff
@@ -38,24 +37,53 @@ public function __construct()
         // It's done inside a middleware closure in order to have
         // the complete request inside the CrudPanel object.
         $this->middleware(function ($request, $next) {
-            $this->crud = app('crud');
+            if (! Backpack::hasCrudController(get_class($this))) {
+                $this->initializeCrud($request);
 
-            $this->crud->setRequest($request);
+                return $next($request);
+            }
 
-            LifecycleHook::trigger('crud:before_setup_defaults', [$this]);
-            $this->setupDefaults();
-            LifecycleHook::trigger('crud:after_setup_defaults', [$this]);
+            $this->triggerControllerHooks();
 
-            LifecycleHook::trigger('crud:before_setup', [$this]);
-            $this->setup();
-            LifecycleHook::trigger('crud:after_setup', [$this]);
+            $this->initialized = true;
```

Comment:
Something's fishy here. I see that we have a function called `initializeCrud()` but we are also doing `$this->initialized=true` outside of it. That is fishy. How can we refactor this bit of code so it's cleaner?